reconcile/utils/git.py

```diff
@@ -25,3 +25,14 @@ def checkout(commit, wd):
     proc.communicate()
     if proc.returncode != 0:
         raise GitError('git checkout failed: {}'.format(commit))
+
+
+def is_file_in_git_repo(file_path):
+    real_path = os.path.realpath(file_path)
+    dir_path = os.path.dirname(real_path)
+    # pylint: disable=consider-using-with
+    DEVNULL = open(os.devnull, 'w')
+    proc = Popen(['git', 'rev-parse', '--is-inside-work-tree'],
+                 cwd=dir_path, stdout=DEVNULL, stderr=DEVNULL)
+    proc.communicate()
```

Comment:
It's easier to just use `subprocess.PIPE` here instead of that `devnull` redirection.

Reply:
as this is a security problem, i'm going to merge it before i start working on tests i will never be able to complete in a timely manner.

Reply:
also note that the only change i made to `ts.dump` is tested, so :shrug:

Reply:
https://github.com/app-sre/qontract-reconcile/pull/2085/commits/ccd0397437ed1de451dc6d9a9c9fa7ab0e0314b4